notebooks_preprocess.py

```diff
@@ -0,0 +1,74 @@
+import glob
+import json
+import os
+import re
+
+
+def format_script_for_cell(path):
+    """Reads and formats a .py file to be inserted into the json for a cell."""
+    cell_content = '\n# Cell content replaced by notebook_preprocess.py\n'
+    with open(path) as f:
+        for line in f:
+            cell_content += line
+    return cell_content
+
+
+def find_load_magics_in_cell(cell):
+    """Find the load magics in a cell and return them as a list."""
+    cell_source = cell['source']
+    load_magics = []
+    for cell_source_line in cell_source:
+        m = re.match('#\s?%load.*', cell_source_line)
+        if m:
+            load_magics.append(m.group())
+    return load_magics
+
+
+def get_cell_content_as_string(cell):
+    """Return the cells source as a single string."""
+    cell_source = cell['source']
+    cell_string = ''
+    for line in cell_source:
+        cell_string += line
+    return cell_string + '\n'
+
+def process_cell(path, cell):
+    """Append the data from the load magics into the cell content."""
+    modified = False
+    # See if there are any load magics used
+    load_magics = find_load_magics_in_cell(cell)
+
+    # Replace the load magics with content from their recpective files
+    for magic_string in load_magics:
+        script_path = os.path.dirname(path) + os.sep + magic_string.split('load ')[1]
```

Comment:
```python
from pathlib import Path
path = Path(path)
script_path = path.parent / magic_string.split('load')[1]
```

If not `pathlib`, then at least use `os.path.join` rather than manually adding `os.sep`